graphqlapi/src/main/java/io/stargate/graphql/schema/graphqlfirst/fetchers/deployed/InsertFetcher.java

```diff
@@ -70,75 +73,105 @@ protected Object get(DataFetchingEnvironment environment, StargateGraphqlContext
             .map(EntityField::getName)
             .orElse(null);
     Keyspace keyspace = context.getDataStore().schema().keyspace(entityModel.getKeyspaceName());
-    Map<String, Object> input = environment.getArgument(model.getEntityArgumentName());
-    Map<String, Object> response = new LinkedHashMap<>();
-    Map<String, Object> cqlValues = buildCqlValues(entityModel, keyspace, input);
-    Collection<ValueModifier> modifiers =
-        cqlValues.entrySet().stream()
-            .map(e -> ValueModifier.set(e.getKey(), e.getValue()))
-            .collect(Collectors.toList());
-
-    Optional<Long> timestamp =
-        TimestampParser.parse(model.getCqlTimestampArgumentName(), environment);
-
-    AbstractBound<?> query =
-        context
-            .getDataStore()
-            .queryBuilder()
-            .insertInto(entityModel.getKeyspaceName(), entityModel.getCqlName())
-            .value(modifiers)
-            .ifNotExists(isLwt)
-            .ttl(model.getTtl().orElse(null))
-            .timestamp(timestamp.orElse(null))
-            .build()
-            .bind();
-
-    context
-        .getAuthorizationService()
-        .authorizeDataWrite(
-            context.getSubject(),
-            entityModel.getKeyspaceName(),
-            entityModel.getCqlName(),
-            TypedKeyValue.forDML((BoundDMLQuery) query),
-            Scope.MODIFY,
-            SourceAPI.GRAPHQL);
-
-    ResultSet resultSet = executeUnchecked(query, buildParameters(environment), context);
-
-    if (responseContainsEntity) {
-      Map<String, Object> entityData;
-      if (entityPrefixInReponse == null) {
-        entityData = response;
+    List<Map<String, Object>> inputs;
+    List<Map<String, Object>> responses = new ArrayList<>();
+    // needed if the return type is a List of Booleans
+    List<Boolean> appliedList = new ArrayList<>();
```

Comment:
It cannot be merged into one `List<Object>` because the logic creates an inner `Map` (`response`) that is added to `responses`:
```
   responses.add(response);
```
so we won't have a flat structure to represent a `List<Boolean>`, but List<Map<...>>`